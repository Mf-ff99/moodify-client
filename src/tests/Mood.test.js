import React from 'react'
import Mood from '../components/mood'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const moodProps = [
    {
        id: 1,
        current_mood: 6,
        note: 'today sucked',
        category_id: 2,
        user_id: 1,
        date_added: '2020-09-04 11:37:30',
    }, {
        id: 2,
        current_mood: 6,
        note: 'today sucked',
        category_id: 2,
        user_id: 1,
        date_added: '2020-09-04 11:37:30',
    }, {
        id: 3,
        current_mood: 6,
        note: 'today sucked',
        category_id: 2,
        user_id: 1,
        date_added: '2020-09-04 11:37:30',
    }, {
        id: 4,
        current_mood: 6,
        note: 'today sucked',
        category_id: 2,
        user_id: 1,
        date_added: '2020-09-04 11:37:30',
    },
]

describe('Mood Component', () => {

    it('renders without crashing', () => {
        shallow(<Mood props={moodProps} />)
    })

    it('renders props approriately', () => {
        const wrapper = shallow(<Mood props={moodProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})