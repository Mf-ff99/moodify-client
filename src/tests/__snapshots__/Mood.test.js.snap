// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mood Component renders props approriately 1`] = `
<Fragment>
  <li
    className="mood-item"
    onClick={[Function]}
  >
    <h2>
      /10
    </h2>
    <h3>
      Invalid Date
    </h3>
    <button
      className="expand-mood-btn"
      onClick={[Function]}
      type="button"
    >
      Expand
    </button>
  </li>
</Fragment>
`;
